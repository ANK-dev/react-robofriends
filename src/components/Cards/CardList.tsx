import React from 'react';
import Card from './Card';
import {IRobot} from '../MainPage/MainPage'

const Cardlist = ({robots}: {robots: IRobot[]}) => {
    return (
        <div>
            {
                robots.map((user, i) => {
                    return (
                        <Card
                            key={robots[i].id}
                            id={robots[i].id}
                            name={robots[i].name}
                            email={robots[i].email}
                        />
                    );
                })
            }
        </div>
    );
}

export default Cardlist;