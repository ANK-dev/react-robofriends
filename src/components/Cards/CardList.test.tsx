import CardList from './CardList';
import React from 'react';
import {shallow} from 'enzyme';

it('expect to render CardList component', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'John Snow',
            username: 'JohnJohn',
            email: 'john@john.com'
        }
    ]
    expect(shallow(<CardList robots={mockRobots}/>)).toMatchSnapshot();
});