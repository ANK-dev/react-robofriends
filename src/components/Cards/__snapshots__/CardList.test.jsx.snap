// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
<div>
  <Card
    email="john@john.com"
    id={1}
    key="1"
    name="John Snow"
  />
</div>
`;
